tests/device_tests/nostr/test_nostr.py

```diff
@@ -0,0 +1,117 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2025 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import json
+import time
+from hashlib import sha256
+
+import pytest
+from ecdsa import SECP256k1, VerifyingKey
+from six import b
+
+from trezorlib import nostr
+from trezorlib.debuglink import TrezorClientDebugLink as Client
+from trezorlib.tools import parse_path
+
+pytestmark = [pytest.mark.altcoin, pytest.mark.models("core")]
+
+# test data from NIP-06: https://github.com/nostr-protocol/nips/blob/master/06.md
+
+LEAD_MONKEY_MNEMONIC = (
+    "leader monkey parrot ring guide accident before fence cannon height naive bean"
+)
+LEAD_MONKEY_PK = "17162c921dc4d2518f9a101db33695df1afb56ab82f5ff3e5da6eec3ca5cd917"
+
+WHAT_BLEAK_MNEMONIC = "what bleak badge arrange retreat wolf trade produce cricket blur garlic valid proud rude strong choose busy staff weather area salt hollow arm fade"
+WHAT_BLEAK_PK = "d41b22899549e1f3d335a31002cfd382174006e166d3e658e3a5eecdb6463573"
+
+
+@pytest.mark.setup_client(mnemonic=LEAD_MONKEY_MNEMONIC)
+def test_sign_event_lead_monkey(client: Client):
```

Comment:
there's a better way to write these tests:
```
LEAD_MONKEY = pytest.mark.setup_client(...)
WHAT_BLEAK = pytest.mark.setup_client(...)

VECTORS = (  # pubkey_hex
    pytest.param("cafecacebeef", mark=LEAD_MONKEY),
    pytest.param(...)
)

@pytest.mark.parametrize("pubkey_hex", VECTORS)
def test_get_pubkey(client, pubkey_hex):
   ...
```

dtto for `sign_event` except i'd prefer _also_ hardcoding the signature bytes, in addition to verifying

Reply:
https://github.com/trezor/trezor-firmware/pull/4436/commits/7f493ea5089762ee62535f37cc63904ebd851d6b#diff-1a61ba923388ca946c730e9cec2ca549a97e355ff7a53673617057d39f95f0f5R44

The signature cannot be hardcoded because every time an event is signed it gets a different signature. That is the reason to use `verify` rather than comparing to a known signature.

Reply:
> The signature cannot be hardcoded because every time an event is signed it gets a different signature.

what is the source of this nondeterminism?

Reply:
It's a feature of ECDSA - a random value is included. This is to prevent deriving of the private key if you have many signatures that you can analyze.

Reply:
that's what RFC 6979 is for -- i was under the impression that this is built into ecdsa signing on library level...?

Reply:
Right... I think that in theory we should be allowed to use the deterministic signatures, but from what I noticed Nostr signatures are usually of the non-deterministic kind. Do you think it should be otherwise? I guess the library can do both, if needed, but isn't the non-deterministic version better?

Reply:
> isn't the non-deterministic version better?

i'm not aware of any argument why it would be.
in our particular case, there's another issue on top: if the nonce is not deterministic, attacks like Dark Skippy can use the signatures to smuggle out secret bits without the user being able to detect it. (with rfc6979 at least you can verify the signature if you also have the seed on the host side)

so please change to deterministic signatures, unless you have a spec explicitly saying you shouldn't do that. (and if so let's see it and talk to R&D about this)

Reply:
I don't think RFC 6979 would work actually because we need Schnorr signatures. So I used BIP-340 to sign and I got deterministic signatures now. 45d4ee0796e573dba4f1eb6bafe1f7662e3c7959